compiler/rustc_trait_selection/src/solve/eval_ctxt/select.rs

```diff
@@ -284,3 +291,130 @@ fn rematch_object<'tcx>(
         ImplSource::Object(ImplSourceObjectData { vtable_base, nested })
     }))
 }
+
+/// The `Unsize` trait is particularly important to coercion, so we try rematch it.
```

Comment:
please add a comment to this and the assemble unsize candidate code that they have to remain in sync